crates/bevy_time/src/time.rs

```diff
@@ -122,11 +148,29 @@ impl Time {
     }
 
     /// The time from startup to the last update in seconds
+    ///
+    /// If you intend to cast this to an `f32` value, note that this value is monotonically increasing,
+    /// and that its precision as an `f32` will noticeably degrade over time (in a matter of hours).
+    /// If that precision loss is unacceptable, you should use [`Time::seconds_since_startup_f32_wrapped`],
+    /// which will return the time from startup modulo a wrapping period.
     #[inline]
     pub fn seconds_since_startup(&self) -> f64 {
         self.seconds_since_startup
     }
 
+    /// The time from startup to the last update, modulo the given wrapping period, in seconds.
+    ///
+    /// Time from startup is a monotonically increasing value and so its precision when read as an `f32`
+    /// will noticeably degrade over time, which causes issues for some uses, e.g. shaders.
+    /// This method avoids noticeable degradation by limiting the values to a much smaller range.
+    ///
+    /// The default wrapping period is one hour.
+    #[inline]
+    pub fn seconds_since_startup_f32_wrapped(&self, duration: WrapDuration) -> f32 {
```

Comment:
Why specify `f32` when there are no alternatives?

Reply:
to make it easier to find if someone just searches for `f32` on `Time` or uses autocomplete since it's a variant of seconds_since_startup but that returns a different type.

Reply:
Fair point. If it weren't for the autocomplete we could have used a doc alias instead.

Maybe, for symmetry we should rename `seconds_since_startup` to `seconds_since_startup_f64`? Keep in mind that it would introduce a breaking change though.

Also, would it be preferrable to have the type specified at the end of the name instead of the middle?

Reply:
The type at the end of the function is a good idea to be more consistent with most api that specify a type. As for modifying the other existing api. I'd prefer waiting on #5752 which refactors all the existing apis.